crates/libcontainer/src/syscall/linux.rs

```diff
@@ -198,6 +198,25 @@ impl Syscall for LinuxSyscall {
         Ok(())
     }
 
+    /// Sets domainname for process (see
+    /// [setdomainname(2)](https://man7.org/linux/man-pages/man2/setdomainname.2.html)).
+    fn set_domainname(&self, domainname: &str) -> Result<()> {
+        let ptr = domainname.as_bytes().as_ptr() as *const c_char;
+        let len = domainname.len();
+        let res = unsafe { setdomainname(ptr, len) };
+        if res == EFAULT {
+            bail!("Failed to set {} as domainname. domainname pointed outside of user address space.", domainname)
+        }
+        if res == EINVAL {
+            bail!("Failed to set {} as domainname. domainname len was negative or too large.", domainname)
+        }
+        if res == EPERM {
+            bail!("Failed to set {} as domainname. the caller did not have the CAP_SYS_ADMIN capability.", domainname)
+        }
```

Comment:
How about using `match` instead of if statements?

Reply:
Modified to match.
And when setdomainname syscall executed errno handling is invalid so fix it.